developing/vcs/README.md

```diff
@@ -2,7 +2,321 @@
 
 ## Overview
 
-Source control has less to do with the vcs system you use and more to do with the stages code goes through on the journey from inside a developers head, tested and merged into the main line, deployed to production and thence in time to being identified as that accursed legacy system that is causing all the problems.
+Source control has less to do with the VCS system you use, and more to do with the stages code goes through on the journey from inside a developer's head, tested and merged into the main line, deployed to production, and thence in time to being identified as that accursed legacy system that is causing all the problems.
+
+## Repo maintenance and hygiene
+
+* Avoid including files in source control that are specific to your development machine or process.
+* Ignore files that are specific to your machine or process in your
+`~/.gitignore_global` as opposed to your repo's `.gitignore`.
+* Delete local and remote feature branches after merging. This can be automated
+on the GitHub side with [Probot](https://probot.github.io/apps/delete-merged-branch/).
+Locally, you can add an alias to your `~/.gitconfig` file. Example:
+
+  ```
+  [alias]
+    fresh = "!git co master && git pull && git branch --merged | grep -v '\\*' | xargs -n 1 git branch -d"
+  ```
+
+  Then instead of doing the typical `git checkout master && git pull`, you would
+  only run `git fresh`.
+
+### GitHub repo settings
+
+* [Protect the master branch](https://help.github.com/en/articles/configuring-protected-branches)
+by turning on the following settings at a minimum:
+
+  * Require pull request reviews before merging
+  * Require status checks to pass before merging
+  * Include administrators
+
+* Under `Settings->Merge button`:
+  * Uncheck all options except for `Allow merge commits` because it's the only
+  one that allows deleting local branches with `-d`. We should never have to use
+  `-D` to delete a branch.
+
+## PR size
+
+Pull requests should be small and focused enough to be reviewable in under 15,
+or ideally under 5, minutes.
+
+## Git workflow
+
+### Fork vs clone
+
+If it's a repo you have write and push access to, clone it. Otherwise, fork it.
+
+### Cloning with SSH vs HTTPS
```

Comment:
I have strong opinions. Use https for read-only repos (ie, where you’re only ever going to clone and pull) and use ssh for everything else. I believe the why is because ssh supports certificates on security keys, and https does not; therefor, we should *not* use https for anything authenticated. However, if you’re just pulling a public repo, https is way more convenient since you don’t need to authenticate anything (unlike ssh for read-only repos).